nemo/collections/tts/data/dataset.py

```diff
@@ -683,6 +697,39 @@ def __getitem__(self, index):
         if SpeakerID in self.sup_data_types_set:
             speaker_id = torch.tensor(sample["speaker_id"]).long()
 
+        gst_ref_audio, gst_ref_audio_length = None, None
+        if GSTRefAudio in self.sup_data_types_set:
+            ref_sample = sample
+            if self.speaker_to_index is not None:
+                if len(self.speaker_to_index[sample["speaker_id"]]) > 1:
+                    ref_pool = self.speaker_to_index[sample["speaker_id"]] - set([index])
+                else:
+                    ref_pool = self.speaker_to_index[sample["speaker_id"]]
+                ref_sample = self.data[random.sample(ref_pool, 1)[0]]
+
+            ref_features = self.featurizer.process(
+                ref_sample["audio_filepath"],
+                trim=self.trim,
+                trim_ref=self.trim_ref,
+                trim_top_db=self.trim_top_db,
+                trim_frame_length=self.trim_frame_length,
+                trim_hop_length=self.trim_hop_length,
+            )
+
+            gst_ref_audio, gst_ref_audio_length = ref_features, torch.tensor(ref_features.shape[0]).long()
+
+        speaker_emb = None
+        if SpeakerEmbedding in self.sup_data_types_set:
+            if self.speaker_embeddings is not None:
+                ref_sample_index = index
+                if self.speaker_to_index is not None:
+                    if len(self.speaker_to_index[sample["speaker_id"]]) > 1:
```

Comment:
This logic needs to be converted into a method as it is being called again in GST feature extraction above.